README.md

```diff
@@ -251,6 +253,86 @@ capability might be supported but not enabled.
 | JSCVT     | Perform Conversion to Match Javascript
 | DCPOP     | Persistent Memory Support
 
+### Custom Features
+The Custom feature source allows the user to define features based on a mix of predefined rules.
+
+#### Rule Types
+
+##### PciId Rule
+The PciId rule allows matching the PCI devices in the system on the following PCI attributes: `class`,`vendor` and
+`device` identifiers.
+
+```yaml
+pciId :
+  class: [<class id>, ...]
+  vendor: [<vendor id>,  ...]
+  device: [<device id>, ...]
+```
+Matching is done by performing a logical OR for elements of an attribute and logical AND between specified attributes for
+each PCI device in the system.
```

Comment:
One more thing (just came to my mind) 😉 Should we explicitly state that all attributes are optional (not all of them need to be specified in a rule), the only (quite logical) requirement is that the user should specify at least one attribute? Or is it obvious from the documentation WDYT?

Reply:
ill explicitly state that the attributes are optional and missing attributes will not partake in the matching process.

there is an example below that depicts that, but i think its worth specifying it here